R/create.boxplot.R

```diff
@@ -62,50 +62,28 @@ create.boxplot <- function(
 		fontface = text.fontface
 		);
 
-	if (!is.null(yat) && length(yat) == 1) {
-		if (yat == 'auto') {
-			out <- auto.axis(unlist(data[toString(formula[[2]])]));
-			data[toString(formula[[2]])] <- out$x;
-			yat <- out$at;
-			yaxis.lab <- out$axis.lab;
-			}
-
-		else if (yat == 'auto.linear') {
-			out <- auto.axis(unlist(data[toString(formula[[2]])]), log.scaled = FALSE);
-			data[toString(formula[[2]])] <- out$x;
-			yat <- out$at;
-			yaxis.lab <- out$axis.lab;
-			}
-
-		else if (yat == 'auto.log') {
-			out <- auto.axis(unlist(data[toString(formula[[2]])]), log.scaled = TRUE);
-			data[toString(formula[[2]])] <- out$x;
-			yat <- out$at;
-			yaxis.lab <- out$axis.lab;
-			}
-		}
+        out <- prep.axis(
+                at = xat,
+                data = unlist(data[toString(formula[[3]])]),
```

Comment:
I'm not sure what `formula` is but `3` looks like a magic number, more examples below

Reply:
Hmm. `formula` is just the `formula` argument to the function. These lines just extract the second half of the argument.

Now that you point this out, I don't think this is 100% bulletproof. It will likely break with any formula with a more complex left side. That problem is probably outside the scope of this PR, though. It would require a refactor with some sort of utility function like `get.formula.args` that accounts for more edge cases.